app/channel.go

```diff
@@ -366,6 +366,12 @@ func (a *App) UpdateChannelPrivacy(oldChannel *model.Channel, user *model.User)
 			return channel, err
 		}
 
+		a.InvalidateCacheForChannel(channel)
+
+		messageWs := model.NewWebSocketEvent(model.WEBSOCKET_EVENT_CHANNEL_UPDATED, channel.TeamId, "", "", nil)
```

Comment:
If you do need to send it to everyone on the team, only send the channel ID not the full channel object

Reply:
Yes, need to send to everyone on the team. Will do send channel ID only and will add test.

Reply:
PR updated, should be good for review.

Reply:
I'm actually leaning towards creating a new WS event specifically for this case, for two reasons:

1. Changing the behaviour of this existing event could break others using the WS for integrations
2. I'd rather deliver the channel as part of WS payload for the 99% of the time where it's a regular update so that the client doesn't have to make a separate fetch after receiving the event. This is also how the rest of our WS events work, so it's nice to maintain consistency.

Thoughts? @lieut-data feel free to chime in as well :)

Sorry for the late response (feel free to ping me on pre-release if that happens).

Reply:
@jwilander Thanks and no problem.
I've updated the PR adding separate "channel_converted" websocket event.